test/api/requests/published-branch.test.js

```diff
@@ -118,6 +118,36 @@ describe('Published Branches API', () => {
             done();
           }).catch(done);
     });
+
+    it('returns a 400 if the access keys are invalid', (done) => {
+      let site;
+      const userPromise = factory.user();
+      const sitePromise = factory.site({
+        users: Promise.all([userPromise]),
+        demoBranch: 'demo',
+      });
+      const cookiePromise = authenticatedSession(userPromise);
+
+      AWSMocks.mocks.S3.listObjectsV2 = (params, callback) => {
+        callback({ status: 403, code: 'InvalidAccessKeyId' }, {});
+      };
+
+      Promise.props({
+        user: userPromise,
+        site: sitePromise,
+        cookie: cookiePromise,
+      }).then((promisedValues) => {
+        site = promisedValues.site;
+
+        return request(app)
+          .get(`/v0/site/${site.id}/published-branch`)
+          .set('Cookie', promisedValues.cookie)
+          .expect(400);
+      }).then((response) => {
+        validateAgainstJSONSchema('GET', '/site/{site_id}/published-branch', 400, response.body);
```

Comment:
We could probably throw and expectation in here to look at the error message.

Reply:
Good call, we do that in other tests, will add!